source/s3_auto_ranged_get.c

```diff
@@ -821,6 +806,17 @@ static void s_s3_auto_ranged_get_request_finished(
                 }
                 /* fall through */
             case AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_GET_OBJECT_WITH_RANGE:
+                if (empty_file_error) {
```

Comment:
we do have logic somewhere to prevent file getting different parts of file if it being actively updated on s3 side. what currently happens if we make a req, and get file is empty error, but then when we get the first part, its been updated on s3 side and is not empty anymore?

Reply:
If we make a request and receive an empty file error, we don't store the ETag at https://github.com/awslabs/aws-c-s3/blob/main/source/s3_auto_ranged_get.c#L630C13-L630C34 since the request failed. So, we try to download the file again with a `partNumber`.
- If the file is still empty, we provide good headers to the user, and the overall request is successful.
- If the file has been updated on the S3 side, we proceed with the normal flow of downloading the first part, storing its ETag, and then parallel downloading the rest of the parts while ensuring that the file has not changed again.